### ISSUE TITLE
Download a new puzzle batch if rating evolves too much

### ISSUE THREAD

**Issue by veloce** [lichess-org/mobile#215]
To ensure puzzle difficulty is always adapted.

We should save the rating at the time the batch is downloaded and then compare from time to time the current rating of the user with that saved rating. If the rating diff goes beyond a certain threshold, a new puzzle batch should be downloaded.

**Comment by braden-godley** [comment]
I totally agree, this is a frustrating experience as someone who mostly plays in mobile. I am currently rated 1825, but I am consistently being given puzzles rated 1550-1600. Theoretically once the batch runs out it will be resolved, but the experience of only getting puzzles that are rated 200 below does kind of suck.

**Comment by ijm8710** [comment]
I def had the same thoughts when my puzzle rating was more chaotic/not as matured when I first joined the site...

I feel a way to manually dump the batch and resync would be helpful as well.
